graylog2-server/src/main/java/org/graylog2/indexer/EventsIndexMapping.java

```diff
@@ -0,0 +1,223 @@
+                                                                .put("index", true)
+                                                                .build())
+                                                        .build())
+                                                .build())
+                                        .add(map()
+                                                .put("fields-typed-boolean", map()
+                                                        .put("path_match", "fields_typed.boolean.*")
+                                                        .put("mapping", map()
+                                                                .put("type", "boolean")
+                                                                .put("doc_values", true)
+                                                                .put("index", true)
+                                                                .build())
+                                                        .build())
+                                                .build())
+                                        .add(map()
+                                                .put("fields-typed-ip", map()
+                                                        .put("path_match", "fields_typed.ip.*")
+                                                        .put("mapping", map()
+                                                                .put("type", "ip")
+                                                                .put("doc_values", true)
+                                                                .put("index", true)
+                                                                .build())
+                                                        .build())
+                                                .build())
+                                         */
+                                        .build())
+                                .put("properties", map()
+                                        .put("id", map()
+                                                .put("type", "keyword")
+                                                .build())
+                                        .put("creator_type", map()
+                                                .put("type", "keyword")
+                                                .build())
+                                        .put("creator_id", map()
+                                                .put("type", "keyword")
+                                                .build())
+                                        .put("timestamp", map()
+                                                .put("type", "date")
+                                                // Use the same format we use for the "message" mapping to make sure we
+                                                // can use the search.
+                                                .put("format", "yyyy-MM-dd HH:mm:ss.SSS")
```

Comment:
As a sidenote: I am wondering if it would make sense to change this to include the timezone in the future. While it does not necessarily introduce bugs to leave it out and use UTC everywhere, it requires everyone indexing a message to adhere to this, while storing the timestamp with the timezone does not.

Reply:
I have been experimenting with several builtin date formats here but the problem is, that our searches won't work at the moment when we use a different date format in ES. :confused: 

So changing this will probably result in a lot of work. ES is storing all dates as UTC internally (it's basically a long) so not sure if it's worth chaning it. But I agree that dates without timezone always look fishy. :wink:

Reply:
Maybe we can address this after replacing the existing search with the views, I think we have a few more options there.